packages/module/src/DataViewTh/DataViewTh.tsx

```diff
@@ -0,0 +1,282 @@
+import {
+  FC,
+  useState,
+  MouseEvent as ReactMouseEvent,
+  TouchEvent as ReactTouchEvent,
+  KeyboardEvent as ReactKeyboardEvent,
+  ReactNode,
+  useCallback,
+  useRef,
+  useEffect
+} from 'react';
+import { Th, ThProps } from '@patternfly/react-table';
+import { Button, getLanguageDirection } from '@patternfly/react-core';
+export interface DataViewThResizableProps {
+  /** Whether the column is resizable */
+  isResizable?: boolean;
+  /** Callback after the column is resized */
+  onResize?: (
+    event: ReactMouseEvent | MouseEvent | ReactKeyboardEvent | KeyboardEvent | TouchEvent,
+    width: number
+  ) => void;
+  /** Width of the column */
+  width?: number;
+  /** Minimum width of the column */
+  minWidth?: number;
+  /** Increment for keyboard navigation */
+  increment?: number;
+  /** Increment for keyboard navigation while shift is held */
+  shiftIncrement?: number;
+  /** Aria label for the resize button */
+  resizeButtonAriaLabel?: string;
```

Comment:
Can we add a console warning for when resizing functionality is enabled but this aria label prop isn't passed in?

I kinda want to lean us also not having a default value for this aria label prop since it would require more context to be taken into account for it (rather than just a consumer being able to use this component a bunch and have a bunch of buttons that all say "Resize column" if that were the default value). Instead it requiring consumers to have to consider that context themselves and pass in "Resize column [name]" or even "Resize column [1|2|3|...]"